frontend/src/components/CreateEvent/CreateEvent.jsx

```diff
@@ -69,6 +93,27 @@ function CreateEvent ({onClose, onCreate, onEdit, initialView, isEdit}) {
                     <p>Date</p>
                     <input type="date" placeholder="mm/dd/yyyy" value={date} onChange={(e) => setDate(e.target.value)} required/>
 
+                    <p>Duration</p>
+                    <input type="number" value={duration} onChange={(e) => setDuration(e.target.value)} placeholder="Duration in 30-min slots" min="1" />
```

Comment:
From a user pov, the duration is pretty unclear. You should provide options of 30 minute increments.